### ISSUE TITLE
Cannot commit or reload packages in Pharo 8

### ISSUE THREAD

**Issue by jecisc** [pharo-vcs/iceberg#1237]
I don't know what is the problem exactly but randomly it happens to me that I cannot commit or reload packages in Pharo 8. 

I get this error:

```
UndefinedObject(Object)>>doesNotUnderstand: #superclassName
UndefinedObject>>doesNotUnderstand: #superclassName
IceClassDefinition>>asMCDefinitionWithoutMetaSide
IceClassDefinition>>removeFromPatcher:
IceMCSnapshotPatchVisitor>>visitRemoval:
IceRemoval>>accept:
[ anIceNode value accept: self ] in IceMCSnapshotPatchVisitor(IceTreeVisitor)>>visitTreeNode: in Block: [ anIceNode value accept: self ]
BlockClosure>>ensure:
IceMCSnapshotPatchVisitor(IceTreeVisitor)>>withNode:do:
IceMCSnapshotPatchVisitor(IceTreeVisitor)>>visitTreeNode:
IceNode>>accept:
[ :each | each accept: self ] in IceMCSnapshotPatchVisitor(IceTreeVisitor)>>visitChildrenOf: in Block: [ :each | each accept: self ]
Array(SequenceableCollection)>>do:
IceNode>>childrenDo:
IceMCSnapshotPatchVisitor(IceTreeVisitor)>>visitChildrenOf:
IceMCSnapshotPatchVisitor>>visitNoModification:
IceNoModification>>accept:
[ anIceNode value accept: self ] in IceMCSnapshotPatchVisitor(IceTreeVisitor)>>visitTreeNode: in Block: [ anIceNode value accept: self ]
BlockClosure>>ensure:
IceMCSnapshotPatchVisitor(IceTreeVisitor)>>withNode:do:
IceMCSnapshotPatchVisitor(IceTreeVisitor)>>visitTreeNode:
IceNode>>accept:
IceGitWorkingCopyUpdateVisitor>>visitPackage:
IcePackageDefinition>>accept:
IceGitWorkingCopyUpdateVisitor>>visitNoModification:
IceNoModification>>accept:
[ anIceNode value accept: self ] in IceGitWorkingCopyUpdateVisitor(IceTreeVisitor)>>visitTreeNode: in Block: [ anIceNode value accept: self ]
BlockClosure>>ensure:
IceGitWorkingCopyUpdateVisitor(IceTreeVisitor)>>withNode:do:
IceGitWorkingCopyUpdateVisitor(IceTreeVisitor)>>visitTreeNode:
```

I don't know how to get a reproducible way yet, but it is really annoying since I cannot commit my changes... :(

**Comment by pavel-krivanek** [comment]
I had the same error when I moved a class from one package to another and wanted to commit

**Comment by jecisc** [comment]
I got a reproducible case.

I took a clean Pharo 8 image. (Pharo-8.0.0+build.283.sha.e9a1fae4c4a0508b4770451dd68daf897daa7c90 (64 Bit)
Unnamed)

I loaded this exact commit of Spec: 

```Smalltalk
Metacello new
	baseline: 'Spec';
	githubUser: 'jecisc' project: 'Spec' commitish: '5e19e86220a299d484d40ef7bf0e21a574df43f5' path: 'src';
	onConflict: [ :e | e useIncoming ];
	onWarningLog;
	load
```

Then I renamed `RadioButtonGroupExample` into `RadioButtonExample`. 

Then I tried to commit, and there Iceberg is throwing me the error. 

I hope this is simple enough to be used to reproduce the bug.

**Comment by juliendelplanque** [comment]
I already encountered this bug as well in the same conditions as @pavel-krivanek

**Comment by jecisc** [comment]
I cannot sync Spec in latest Pharo 8 because it happens to me and there is too much problems for me to fix by editing the text files :(

**Comment by jecisc** [comment]
Workaround:

```IceClassDefinition>>removeFromPatcher: aMCPatcher 
	
	"If the meta side is removed, we should modify the class definition to not include the meta-parts such as class-instance-variables"
	mcDefinition ifNotNil: [isMeta ifTrue: [ 
		^ aMCPatcher
			modifyDefinition: self asMCDefinition
			to: self asMCDefinitionWithoutMetaSide
	].
	
	aMCPatcher removeDefinition: self asMCDefinition ]
```

**Comment by guillep** [comment]
A small example we did to reproduce it with Pavel:

 - Create class #A
 - Add class side method #foo
 - commit
 - remove #A
 - commit
=> boom

**Comment by guillep** [comment]
We fixed + tested this with @pavel-krivanek on a130d0d9ca905187f7bf7e4d1fefc1791804269d
